tests/components/androidtv/test_remote.py

```diff
@@ -0,0 +1,147 @@
+"""The tests for the androidtv remote platform."""
+
+from typing import Any
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.components.androidtv.const import (
+    CONF_TURN_OFF_COMMAND,
+    CONF_TURN_ON_COMMAND,
+)
+from homeassistant.components.remote import (
+    ATTR_NUM_REPEATS,
+    DOMAIN as REMOTE_DOMAIN,
+    SERVICE_SEND_COMMAND,
+)
+from homeassistant.const import (
+    ATTR_COMMAND,
+    ATTR_ENTITY_ID,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+)
+from homeassistant.core import HomeAssistant
+
+from . import patchers
+from .common import (
+    CONFIG_ANDROID_DEFAULT,
+    CONFIG_FIRETV_DEFAULT,
+    SHELL_RESPONSE_OFF,
+    SHELL_RESPONSE_STANDBY,
+    setup_mock_entry,
+)
+
+from tests.common import MockConfigEntry
+
+
+def _setup(config: dict[str, Any]) -> tuple[str, str, MockConfigEntry]:
+    """Prepare mock entry for the media player tests."""
+    return setup_mock_entry(config, REMOTE_DOMAIN)
+
+
+async def _test_service(
+    hass: HomeAssistant,
+    entity_id,
+    ha_service_name,
+    androidtv_method,
+    additional_service_data=None,
+    return_value=None,
+) -> None:
+    """Test generic Android media player entity service."""
+    service_data = {ATTR_ENTITY_ID: entity_id}
+    if additional_service_data:
+        service_data.update(additional_service_data)
+
+    androidtv_patch = (
+        "androidtv.androidtv_async.AndroidTVAsync"
+        if "android" in entity_id
+        else "firetv.firetv_async.FireTVAsync"
+    )
+    with patch(
+        f"androidtv.{androidtv_patch}.{androidtv_method}", return_value=return_value
+    ) as service_call:
```

Comment:
Isn't the naming here confusing, the patch is not the service call, it's the API endpoint we expect to be called by the service call?

Reply:
Yes, fixed